internal/providers/terraform/aws/cloudfront_distribution.go

```diff
@@ -1,527 +1,38 @@
-	var paidQuantity *decimal.Decimal
-	if u != nil && !u.IsEmpty("monthly_invalidation_requests") {
-		usageAmount := u.Get("monthly_invalidation_requests").Int()
-		if usageAmount < 1000 {
-			freeQuantity = decimalPtr(decimal.NewFromInt(usageAmount))
-		} else {
-			freeQuantity = decimalPtr(decimal.NewFromInt(1000))
-			paidQuantity = decimalPtr(decimal.NewFromInt(usageAmount - 1000))
-		}
-	}
-
-	costComponents := []*schema.CostComponent{
-		{
-			Name:            "Invalidation requests (first 1k)",
-			Unit:            "paths",
-			UnitMultiplier:  decimal.NewFromInt(1),
-			MonthlyQuantity: freeQuantity,
-			ProductFilter: &schema.ProductFilter{
-				VendorName: strPtr("aws"),
-				Service:    strPtr("AmazonCloudFront"),
-				AttributeFilters: []*schema.AttributeFilter{
-					{Key: "usagetype", Value: strPtr("Invalidations")},
-				},
-			},
-			PriceFilter: &schema.PriceFilter{
-				StartUsageAmount: strPtr("0"),
-			},
-		},
-	}
-
-	if paidQuantity != nil {
-		costComponents = append(costComponents, &schema.CostComponent{
-			Name:            "Invalidation requests (over 1k)",
-			Unit:            "paths",
-			UnitMultiplier:  decimal.NewFromInt(1),
-			MonthlyQuantity: paidQuantity,
-			ProductFilter: &schema.ProductFilter{
-				VendorName: strPtr("aws"),
-				Service:    strPtr("AmazonCloudFront"),
-				AttributeFilters: []*schema.AttributeFilter{
-					{Key: "usagetype", Value: strPtr("Invalidations")},
-				},
-			},
-			PriceFilter: &schema.PriceFilter{
-				StartUsageAmount: strPtr("1000"),
-			},
-		})
-	}
-
-	return costComponents
-}
-
-func encryptionRequests(d *schema.ResourceData, u *schema.UsageData) *schema.CostComponent {
-	if d.IsEmpty("default_cache_behavior.0.field_level_encryption_id") {
-		return nil
-	}
-
-	var quantity *decimal.Decimal
-	if u != nil && !u.IsEmpty("monthly_encryption_requests") {
-		quantity = decimalPtr(decimal.NewFromInt(u.Get("monthly_encryption_requests").Int()))
-	}
-
-	return &schema.CostComponent{
-		Name:            "Field level encryption requests",
-		Unit:            "10k requests",
-		UnitMultiplier:  decimal.NewFromInt(10000),
-		MonthlyQuantity: quantity,
-		ProductFilter: &schema.ProductFilter{
-			VendorName: strPtr("aws"),
-			Service:    strPtr("AmazonCloudFront"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "requestDescription", Value: strPtr("HTTPS Proxy requests with Field Level Encryption")},
-				{Key: "location", Value: strPtr("Europe")},
-			},
-		},
-	}
-}
-
-func realtimeLogs(d *schema.ResourceData, u *schema.UsageData) *schema.CostComponent {
-	if d.IsEmpty("logging_config.0.bucket") {
-		return nil
-	}
-
-	var quantity *decimal.Decimal
-	if u != nil && !u.IsEmpty("monthly_log_lines") {
-		quantity = decimalPtr(decimal.NewFromInt(u.Get("monthly_log_lines").Int()))
+	var originShieldRegion string
+	if !d.IsEmpty("origin.0.origin_shield.0.origin_shield_region") {
+		originShieldRegion = d.Get("origin.0.origin_shield.0.origin_shield_region").String()
```

Comment:
```suggestion
       originShieldRegion := d.Get("origin.0.origin_shield.0.origin_shield_region").String()
```
This defaults to an empty string either way so no need for the `IsEmpty` check

Reply:
Updated 👍